gui/src/logger.rs

```diff
@@ -36,6 +39,7 @@ pub struct Logger {
         Registry,
     >,
     level_handle: reload::Handle<filter::LevelFilter, Registry>,
+    receiver: crossbeam_channel::Receiver<String>,
```

Comment:
I believe it should be removed. 
the layer can be added in `set_running_mode` and the receiver returned as Result of the method

Reply:
If not possible, change it in Option<> and use take to pass it to loader.
The ideal would be that this Receiver is dropped once loader changed its state.

Reply:
When adding it in `set_running_mode`, I'm not sure how to make it active without calling `init` as is done in `setup`. The subscriber returns a message that it's trying to read from a disconnected channel (and this message is shown in the GUI on the loader screen).

The second approach using `take` seems to work in terms of making the channel become disconnected once the loader has completed (or shortly thereafter).

Reply:
We might be able to create an `mpsc` channel inside the iced subscription closure itself as in this example:
https://github.com/iced-rs/iced/blob/dd249a1d11c68b8fee1828d58bae158946ee2ebd/futures/src/subscription.rs#L286-L355
Then the channel should be dropped once the loader completes.

We would pass the sender back to the loader and would just need a way to subscribe to log events while the loader is running. Perhaps using the layer with https://docs.rs/tracing/latest/tracing/dispatcher/fn.set_default.html.